### PR TITLE
FIX: More bugs in Manual Deploy Workflow

### PR DESCRIPTION
**Pull request by mzappitello**
* Inputs need to be passed in from the workflow dispatch to the base workflow as booleans, so use a fromJson function to flip them.
* Rename the env-name input to environment so that it matches with the workflow dispatch name.
* Pass the environment string, not the name of the variable.

* Add prod to list of environments we can change the task count for.

### CODE REVIEW

File: .github/workflows/change_task_count.yaml [hunk 1]
```diff
@@ -9,6 +9,7 @@ on:
         options:
           - dev
           - staging
+          - prod
```
Comment (mzappitello) [inline .github/workflows/change_task_count.yaml:12]: we can change the task count for prod now.

File: .github/workflows/deploy-base.yaml [hunk 2]
```diff
@@ -8,7 +8,7 @@
 on:
   workflow_call:
     inputs:
-      env-name:
+      environment:
```
Comment (mzappitello) [inline .github/workflows/deploy-base.yaml:11]: rename this input, with cascading changes in the deploy workflows that call `deploy-base.yaml`

File: .github/workflows/manual-deploy.yaml [hunk 3]
```diff
@@ -25,13 +25,14 @@ on:
...
       group: github.event.inputs.environment
     uses: ./.github/workflows/deploy-base.yaml
     with:
-      env-name: github.event.inputs.environment
-      deploy-ingestion: github.envent.inputs.deploy-ingestion
-      deploy-rail-pm: github.envent.inputs.deploy-rail-pm
-      deploy-tableau-publisher: github.envent.inputs.deploy-tableau-publisher
+      # pass the inputs from the workflow dispatch through to the deploy base. the booleans are
+      # converted to strings, so flip them back using fromJson function
+      environment: ${{ github.event.inputs.environment }}
```
Comment (mzappitello) [inline .github/workflows/manual-deploy.yaml:34]: i was passing the string of the variable name before 😵‍💫

File: .github/workflows/manual-deploy.yaml [hunk 4]
```diff
@@ -25,13 +25,14 @@ on:
...
+      deploy-ingestion: ${{ fromJson(github.event.inputs.deploy-ingestion) }}
+      deploy-rail-pm: ${{ fromJson(github.event.inputs.deploy-rail-pm) }}
+      deploy-tableau-publisher: ${{ fromJson(github.event.inputs.deploy-tableau-publisher) }}
```
Comment (mzappitello) [inline .github/workflows/manual-deploy.yaml:37]: using `fromJson` converts these from strings back to bools.

Comment (rymarczy) [reply .github/workflows/manual-deploy.yaml:37]: would you want the `fromJson` on the inside or the outside of `${{..}}` here?

**Review by rymarczy — APPROVED** [review]
One question, but otherwise LGTM

File: .github/workflows/manual-deploy.yaml (hunk 4 above)
Comment (mzappitello) [reply .github/workflows/manual-deploy.yaml:37]: expressions go inside.

https://docs.github.com/en/actions/learn-github-actions/expressions#example-returning-a-json-data-type

i still find all of the syntax rules a bit hard to follow, but i'm sure they stem from some weird backend interpreter they have to use.
